test/functional/test_framework/test_node.py

```diff
@@ -480,7 +480,10 @@ def add_p2p_connection(self, p2p_conn, *, wait_for_verack=True, **kwargs):
             #
             # So syncing here is redundant when we only want to send a message, but the cost is low (a few milliseconds)
             # in comparision to the upside of making tests less fragile and unexpected intermittent errors less likely.
-            p2p_conn.sync_with_ping()
+            #
+            # Connections may opt out by passing `sync_with_ping=False`.
```

Comment:
Instead of explaining how to opt out, it would help to explain *why* someone would want to opt out. Is this a performance improvement? I yes, then it should be quantified.

Otherwise it seem like added code-complexity that makes tests more fragile and harder to read, write and maintain.

Reply:
Good idea. Pushed a new version.